src/main/java/com/linecorp/armeria/client/thrift/THttpClientDelegate.java

```diff
@@ -101,8 +103,13 @@ public ThriftReply execute(ClientRequestContext ctx, ThriftCall call) throws Exc
         }
 
         try {
+
             final TMemoryBuffer outTransport = new TMemoryBuffer(128);
-            final TProtocol tProtocol = protocolFactory.getProtocol(outTransport);
+            TProtocol tProtocol = protocolFactory.getProtocol(outTransport);
+            if (ctx.options().tMultiplex()) {
+                final String serviceName = this.path.substring(1);
+                tProtocol = new TMultiplexedProtocol(tProtocol, serviceName);
```

Comment:
You don't really need to use `TMultiplexedProtocol` for this purpose. You can just prepend service name to the method name:

```java
method = serviceName + ':' + method
```